File: clinical/src/main/java/com/hartwig/actin/clinical/feed/questionnaire/LesionData.java
package com.hartwig.actin.clinical.feed.questionnaire;

import static com.hartwig.actin.clinical.feed.questionnaire.QuestionnaireCuration.toOption;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

class LesionData {

    @Nullable
    private final Boolean present;
    @Nullable
    private final Boolean active;

    public LesionData(@Nullable final Boolean present, @Nullable final Boolean active) {
        this.present = present;
        this.active = active;
    }

    @NotNull
    static LesionData fromString(@NotNull String presentInput, @NotNull String activeInput) {
        Boolean present = toOption(presentInput);
        Boolean active = null;
        if (present != null) {
            Boolean activeOption = toOption(activeInput);
            if (activeOption != null) {
                active = present ? activeOption : false;
            }
        }

        return new LesionData(present, active);
    }

    @Nullable
    public Boolean present() {
        return present;
    }

    @Nullable
    public Boolean active() {
        return active;
    }
}

File: clinical/src/main/kotlin/com/hartwig/actin/clinical/feed/questionnaire/LesionData.kt
package com.hartwig.actin.clinical.feed.questionnaire

internal class LesionData(private val present: Boolean?, private val active: Boolean?) {
    fun present(): Boolean? {
        return present
    }

    fun active(): Boolean? {
        return active
    }

    companion object {
        fun fromString(presentInput: String, activeInput: String): ValidatedQuestionnaireCuration<LesionData> {
            val present = curateQuestionnaireOption(presentInput)
            val active = if (present.curated != null) curateQuestionnaireOption(activeInput)
            else ValidatedQuestionnaireCuration(null)

            return ValidatedQuestionnaireCuration(
                LesionData(present.curated, active.curated),
                present.errors + active.errors
            )
        }

        private fun curateQuestionnaireOption(input: String): ValidatedQuestionnaireCuration<Boolean> =
            ValidatedQuestionnaireCuration(QuestionnaireCuration.toOption(input).curated)

    }
}